### PR TITLE
Added hikaricp support

### PR DESCRIPTION
**Pull request by tokuhirom**

### CODE REVIEW

File: simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    this.labelValues = Collections.singletonList(poolName);
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    return Arrays.asList(
+            createSample("hikaricp_active_connections", "Active connections", poolStats.getActiveConnections()),
+            createSample("hikaricp_idle_connections", "Idle connections", poolStats.getIdleConnections()),
+            createSample("hikaricp_pending_threads", "Pending threads", poolStats.getPendingThreads()),
+            createSample("hikaricp_total_connections", "Total connections", poolStats.getTotalConnections())
```
Comment (brian-brazil) [inline simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java]: Is this current connections, or connections created over all time?

File: simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPMetricsTracker.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            .labelNames("pool")
+            .help("Connection timeout count")
+            .register();
+
+    this.counter = counter.labels(poolName);
+  }
+
+  @Override
+  public void recordConnectionAcquiredNanos(long elapsedAcquiredNanos) {
+    super.recordConnectionAcquiredNanos(elapsedAcquiredNanos);
```
Comment (brian-brazil) [inline simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPMetricsTracker.java]: Why not capture these in a summary?

**Comment by brian-brazil** [comment]
Have you asked hikaricp about adding direct instrumentation? That'd be much simpler for users as they'd get all this without any configuration, and there's already dropwizard integration.

File: simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPMetricsTracker.java (hunk 2 above)
Comment (tokuhirom) [inline simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPMetricsTracker.java]: Oh. I don't know about summary. I added code to take these metrics.

**Comment by tokuhirom** [comment]
https://github.com/brettwooldridge/HikariCP/issues/321
HikariCP's author was rejected the feature request :(

File: simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java (hunk 1 above)
Comment (tokuhirom) [inline simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java]: It's the number of current connections.

**Comment by brian-brazil** [comment]
I don't read that as a rejection, I see it that a PR would be considered. It's probably worth clarifying.

**Comment by tokuhirom** [comment]
Oops. I miss understood. I'll try to send p-r to hikaricp repo.

File: simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java (hunk 1 above)
Comment (brian-brazil) [inline simpleclient_hikaricp/src/main/java/io/prometheus/client/hikaricp/HikariCPCollector.java]: I'd suggest removing the `total_` from the name then, and clarifying the docstring.

**Comment by tokuhirom** [comment]
I sen't p-r to prometheus repo.

**Comment by brian-brazil** [comment]
That's in the app itself now.

**Pull request closed without merging**